api/base/views.py

```diff
@@ -129,7 +129,7 @@ def get_serializer_context(self):
         context.update({
             'enable_esi': (
                 is_truthy(self.request.query_params.get('esi', django_settings.ENABLE_ESI)) and
-                self.request.accepted_media_type in django_settings.ESI_MEDIA_TYPES
+                self.request.accepted_renderer.media_type in django_settings.ESI_MEDIA_TYPES
```

Comment:
On the plus side, even if this had been an obscure error and not one that hit you in the face immediately on login, it would have (and was) caught by integration tests. But yes, more of these, and more maintenance on them, would be awesome.